src/plugins/gimbal/include/plugins/gimbal/gimbal.h

```diff
@@ -84,6 +84,32 @@ class Gimbal : public PluginBase {
      */
     void set_pitch_and_yaw_async(float pitch_deg, float yaw_deg, result_callback_t callback);
 
+    /** 
+     * @brief Set gimbal mode (synchronous).
+     *
+     * This sets the desired mode of a gimbal.
+     * The function will return when the command is accepted, however, it might
+     * take the gimbal longer to actually be set to the new angles.
+     *
+     * @param gimbal_mode The mode to be set. Either lock or follow.
+     * Lock will fix the gimbal poiting to a direction. Follow will follow the vehicle heading.
```

Comment:
Please document that this is for yaw.

Reply:
Also, given this can change the behavior of the existing gimbal setting, make sure add a comment there on how the yaw angle is interpreted differently depending on the mode.

Reply:
Can you elaborate. What do you mean by the Yaw angle?

Reply:
I mean the angle that you pass to `set_pitch_and_yaw`, it's different given the mode, right?

Reply:
So, if I understood your questions correctly, the angle yaw input will be the same for both modes, which is `param7 = 2` from options: `(0 = angle body frame, 1 = angular rate, 2 = angle absolute frame)`. What will change is the `stabilize yaw` between `(1 = yes, 0 = no)` 
0 = Yaw Follow
1 = Yaw Lock

Let me know if you meant something different by angle.

Reply:
But doesn't the gimbal yaw angle mean something else? Basically in lock mode 0 will point North, and in follow mode it will point forward.
Therefore, the yaw variable is relative to the world frame for lock, and relative to the vehicle for follow frame.